### PR TITLE
Add Course Detail Components

### PR DESCRIPTION
**Pull request by mathewscullin**
## Overview

Add components for the course detail view page.

## Changes Made

Implemented 3 new files and corresponding CSS.

## Test Coverage

Test by running the web app, ensuring styling matches designs.

## Next Steps (delete if not applicable)

Launch this view using a router and give functionality to track/remove buttons.

## Screenshots (delete if not applicable)

<details>

  <summary>Component Screen Shot</summary>

Screen Shot 2021-05-09 at 6 02 25 PM

  

</details>

### CODE REVIEW

File: src/components/DetailCourse/DetailCourseView.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    [pass, fail] = partition(fail, (s: Section) => s.section.substring(0, 3) == "DIS");
+    if (pass.length > 0) {
+        cards.push(<DetailSectionCard sectionType="Section" sections={pass}/>)
+    }
+    [pass, fail] = partition(fail, (s: Section) => s.section.substring(0, 3) == "LAB");
+    if (pass.length > 0) {
+        cards.push(<DetailSectionCard sectionType="Lab" sections={pass}/>)
+    }
+    if (fail.length > 0) {
+        cards.push(<DetailSectionCard sectionType="Lab" sections={fail}/>)
```
Comment (mathewscullin) [inline src/components/DetailCourse/DetailCourseView.tsx:36]: This string should be `Other` will change that before merging.

File: src/components/DetailCourse/DetailCourseCard.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+import './DetailCourseCard.css'
+
+import { Section } from '../../types'
+
+import DetailSection from './DetailCourseSectionCell'
```
Comment (haiyingweng) [inline src/components/DetailCourse/DetailCourseCard.tsx:7]: i think should be `DetailCourseCardCell`

File: src/components/DetailCourse/DetailCourseCardCell.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    height: 30px;
+  }
+
+  .description-label {
+    vertical-align: middle;
+    color: #000000;
+    font-weight: 600;
+    font-size: 14px;
+    margin-left:8px;
+  }
```
Comment (haiyingweng) [inline src/components/DetailCourse/DetailCourseCardCell.css:20]: the formatting for all the css files is a little off, there're extra two spaces

File: src/components/DetailCourse/DetailCourseCardCell.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+
+import './DetailCourseSectionCell.css'
```
Comment (haiyingweng) [inline src/components/DetailCourse/DetailCourseCardCell.tsx:3]:
```suggestion
import './DetailCourseCardCell.css'
```

File: src/components/DetailCourse/DetailCourseCardCell.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      {
+        !section.isTracking
+          ? <button className="track-button">TRACK</button>
+          : null
+      }
+      {
+        section.isTracking
+          ? <button className="remove-button">REMOVE</button>
+          : null
+      }
```
Comment (haiyingweng) [inline src/components/DetailCourse/DetailCourseCardCell.tsx:49]: would we be able to combine these two to something like 
```javascript
section.isTracking
      ? <button className="remove-button">REMOVE</button>
      : <button className="track-button">TRACK</button>
```

**Review by haiyingweng — APPROVED** [review]
looks fire 🚀🚀

**Pull request merged**